client/src/main/java/com/vesoft/nebula/client/graph/ConnectionContext.java

```diff
@@ -0,0 +1,104 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+package com.vesoft.nebula.client.graph;
+
+/**
+ * some props about connection
+ */
+public class ConnectionContext {
+    private String hosts;
+    private String user;
+    private String password;
+    private String space;
+    private int timeout;
+    private int connectionRetry;
+    private int executionRetry;
+    private int failoverRetry;
+    private int poolSize;
+
+    ConnectionContext withHosts(String hosts) {
+        this.hosts = hosts;
+        return this;
+    }
+
+    ConnectionContext withUser(String user) {
+        this.user = user;
+        return this;
+    }
+
+    ConnectionContext withPassword(String password) {
```

Comment:
If the password is an empty string ... we should support more testing about the argument.